app/controllers/application_controller.rb

```diff
@@ -6,4 +6,8 @@ class ApplicationController < ActionController::Base
   def after_sign_out_path_for(_resource_or_scope)
     new_user_session_path
   end
+
+  def routing_error
+  render :file => "#{Rails.root}/public/404.html",  :status => 404
```

Comment:
Use modern syntax.  Improve indentation of code.
`file: "#{Rails.root}/public/404.html",  status: 404`